FrameworkSystem/DB/DynamicMonitoringDB.py

```diff
@@ -0,0 +1,182 @@
+"""
+Class and utilities for managing dynamic monitoring logs in Elasticsearch
+"""
+
+__RCSID__ = "$Id$"
+
+import datetime
+from DIRAC.Core.Utilities.ElasticSearchDB import ElasticSearchDB
+from DIRAC import gLogger, gConfig, S_OK, S_ERROR
+
+class DynamicMonitoringDB( object ):
+  """
+  This class manages logs stored in Elasticsearch
+  """
+
+  def __init__( self, address, port ):
+    self.esDB = ElasticSearchDB( address, port )
+    self.docType = 'ComponentMonitoring'
+
+  def _dictToQuery( self, matchFieldsDict ):
+    """
+    Transforms a given dictionary with the fields to match into a query dictionary
+    :param dict matchFieldsDict: Dictionary with the fields that the 'match' part of the query should contain
+    :return: Returns the query dictionary to be passed to the elasticsearch API library
+    """
+    query = { 'bool': { 'must': [] } }
+    for key in matchFieldsDict:
+      query[ 'bool' ][ 'must' ].append( { 'match': { key : matchFieldsDict[ key ] } } )
+
+    return query
+
+  def insertLog( self, log ):
+    """
+    Creates a new log entry in ElasticSearch
+    :param dict log: Should be a dictionary containing the logging information to be stored
+    """
+    return self.esDB.addToIndex( '%s_index' % self.docType.lower(), self.docType, log )
+
+  def getLastLog( self, host, component, maxAge = 10 ):
+    """
+    Retrieves the last log for the given component
+    :param str host: The name of the host where the component is installed
+    :param str component: The name of the component in the form system_component
+    :param int maxAge: Tells the method how many days in the past to look for the log. If after reaching maxAge days, no logs have been found, an error will be returned
+    :return: The latest log stored in the database for the given component
+    """
+    date = datetime.datetime.utcnow()
+
+    for i in range( maxAge ):
+      indexName = '%s_index-%s' % ( self.docType.lower(), date.strftime( '%Y-%m-%d' ) )
+
+      result = self.esDB.checkIndex( indexName )
+      if result[ 'OK' ]:
+        if result[ 'Value' ]:
+          result = self.esDB.query( indexName, { 'query': \
+            { 'bool' : { 'must': [ { 'match': { 'host': host } }, { 'match': { 'component': component } } ] } } \
+            , 'sort': { 'timestamp': { 'order': 'desc' } }, 'size': 1 } )
+          if result[ 'OK' ]:
+            if len( result[ 'Value' ][ 'hits' ][ 'hits' ] ) >= 1:
+              return S_OK( [ result[ 'Value' ][ 'hits' ][ 'hits' ][0] ] )
+          else:
+            return result
+      else:
+        return result
+
+      date = date - datetime.timedelta( days = 1 )
+
+    return S_ERROR( 'No logs for this component found' )
```

Comment:
Please "standardize" the error message (here is "No logs for this component found", later "No logs found for the component" ...